tas-client/src/tas-client/FeatureProvider/TasApiFeatureProvider.ts

```diff
@@ -38,7 +38,48 @@ export class TasApiFeatureProvider extends FilteredFeatureProvider {
         }
 
         //axios webservice call.
-        let response: AxiosResponse<TASFeatureData> = await this.httpClient.get({ headers: headers });
+        let response: AxiosResponse<TASFeatureData> | undefined;
+
+        
+        try {
+            /**
+             * As mentioned in the axios docs:
+             * Axios Promise will handle handle the catch
+             * section of promises whenever a request is not succesful.
+             * https://axios-http.com/docs/handling_errors
+             */
+            response = await this.httpClient.get({ headers: headers });
+        } catch(error) {
+            const axiosError = error as AxiosError;
+            const properties: Map<string, string> = new Map();
+            if (axiosError.response) {
+                // The request was made and the server responded with a status code
+                // that falls out of the range of 2xx
+                properties.set("RESPONSE_DATA", axiosError.response.data);
```

Comment:
I'm worried that this could potentially contain sensitive information, e.g., an IP address. Could we leave this out for now until we
(1) Have need of it
(2) Can find a way to make sure it doesn't contain sensitive info/we sanitize it correctly

Reply:
Same concern for `RESPONSE_HEADERS`

Reply:
Agree, we can leave those out for now.
I think if we need them in the future, we could implement some redact process to remove paths and PII from the strings if needed.

Reply:
FWIW, the `@vscode/extension-telemetry` package (which all VSCode extensions use for telemetry) passes everything through a handful of filters. I _think_ it would catch IP addresses. However, there's still a lot of other things that could leak through, so I agree with @daytonellwanger's concern.

Reply:
I've removed all possible PII data and I'm just sending the type of error that ocurred.